docs/field-details.asciidoc

```diff
@@ -6864,3 +6893,339 @@ example: `Critical`
 // ===============================================================
 
 |=====
+
+[[ecs-x509]]
+=== x509 Certificate Fields
+
+These fields contain x509 certificate metadata.
+
+==== x509 Certificate Field Details
+
+[options="header"]
+|=====
+| Field  | Description | Level
+
+// ===============================================================
+
+| x509.alternative_names
+| List of subject alternative names (SAN). Name types vary by certificate authority and certificate type but commonly contain IP addresses, DNS names (and wildcards), and email addresses.
+
+type: keyword
+
+
+
+example: `*.elastic.co`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.common_name
+| Common name (CN) of issuing certificate authority.
+
+type: keyword
+
+
+
+example: `DigiCert SHA2 High Assurance Server CA`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.country
+| Country (C) code
+
+type: keyword
+
+
+
+example: `US`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.distinguished_name
+| Distinguished name (DN) of issuing certificate authority.
+
+type: keyword
+
+
+
+example: `C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert SHA2 High Assurance Server CA`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.locality
+| Locality name (L)
+
+type: keyword
+
+
+
+example: `Mountain View`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.organization
+| Organizational units (O) of issuing certificate authority.
+
+type: keyword
+
+
+
+example: `DigiCert Inc`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.organizational_unit
+| List of organizational units (OU) of issuing certificate authority.
+
+type: keyword
+
+
+
+example: `www.digicert.com`
+
+| extended
+
+// ===============================================================
+
+| x509.issuer.state_or_province
+| State or province name (ST, S, or P)
+
+type: keyword
+
+
+
+example: `California`
+
+| extended
+
+// ===============================================================
+
+| x509.not_after
```

Comment:
Could we pick less confusing fields for valid dates? not_before, not_after seems unnecessarily irritating to deal with (even if they are 100% accurate by rfc)

- x509.start_date
- x509.end_date
or 
- x509.valid.start_date
- x509.valid.end_date

other than that it looks great

Reply:
This is a fair point. I ran through some ideas and looked at terms used in browsers and other libraries. I think "expiration" best captures the semantics of `not_after`.

What do you think about this?

- `x509.valid_date`
- `x509.expire_date`

I think `x509.issued` for `not_before` would be nice, but there's enough "untruth" in there to make me uncomfortable.

Reply:
Updated to the proposed names

Reply:
Yeah I like the clearer names 👍  Resolving this.